frontend-ssr/redux/exercise-submissions/exercise-submissions.reducer.ts

```diff
@@ -0,0 +1,54 @@
+import { SubmissionState } from './types';
+import { EXERCISE_SUBMISSIONS } from './exercise-submissions.actions';
+
+const initialState: SubmissionState = {
+  submissions: [],
+};
+export const submissionReducer = (state = initialState, action: { type: string; payload: any;}) => {
+  switch (action.type) {
+    case EXERCISE_SUBMISSIONS.ADD_SUBMISSION: {
+      const { index, submission } = action.payload;
+      const newArray = [...state.submissions];
+      newArray.splice(index, 0, submission);
+      if (Array.isArray(submission)) {
+        submission.map((item) => ({
+          newArray,
+          submissions: [...state.submissions, item],
+        }));
+      }
+      return {
+        newArray,
+        submissions: [...state.submissions, submission],
+      };
+    }
+    case EXERCISE_SUBMISSIONS.REMOVE_SUBMISSION: {
+      const { id } = action.payload;
+      if (Array.isArray(id)) {
```

Comment:
Did you test this function when sending an Array It doesn't seem to be doing anything...